### PR TITLE
Better management of targets in build_examples.py, support -rpc when building all

### PR DESCRIPTION
**Pull request by andy31415**
#### Problem
Target combinations in build_examples is not always obvious. RPC build is controller by a separate flag.

#### Change overview
Use a `--target` argument that contains the tripet (or more) like `linux-x64-chip-tool` or `tizen-arm-light`
Updated vscode tasks to be somewhat easier to use and list all current available targets
have a 'targets' command that lists all available target combinations

#### Testing
Updated unit tests
Manually validated various builds

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

**Comment by andy31415** [comment]
@Damian-Nordic @hawk248 @jelderton @jepenven-silabs @jmartinez-silabs @LuDuda @msandstedt ?

File: examples/lighting-app/telink/Readme.md [hunk 1]
```diff
@@ -26,7 +26,7 @@
 1. Run build script:
 
     ```
-    ./scripts/build/build_examples.py --app light --platform telink build
+    ./scripts/build/build_examples.py --target telink-arm-light build
```
Comment (Damian-Nordic) [inline examples/lighting-app/telink/Readme.md:29]: Did you meawn `telink-tlsr9518adk80d-light`?

File: scripts/build/README.md [hunk 2]
```diff
@@ -32,34 +32,39 @@ requirement.
 
 Usage examples:
 
-1. Compiles the Lock app on all supported platforms
+1. Compiles All targets
```
Comment (Damian-Nordic) [inline scripts/build/README.md:35]:
```suggestion
1. Compiles all targets
```

File: scripts/build/build/targets.py [hunk 3]
```diff
@@ -12,98 +12,150 @@
...
+        the specified build)
+    """
+
+    def __init__(self, name, builder_class, **kwargs):
+        self.name = name
+        self.builder_class = builder_class
+        self.create_kw_args = kwargs
+
+    def Extend(self, suffix, **kargs):
+        """Creates a clone of the current object extendint it build parameters.
```
Comment (Damian-Nordic) [inline scripts/build/build/targets.py:42]:
```suggestion
        """Creates a clone of the current object extending its build parameters.
```

**Review by msandstedt — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]

**Review by wootak-jung — APPROVED** [review]

**Pull request merged**